src/app/testing/backend-service-stub.ts

```diff
@@ -37,7 +38,32 @@ export class BackendServiceStub extends BaseBackendService {
     });
   }
 
-  createPoem(poemName: string, poemText: string, generated: boolean) {
-    // Not yet implemented
+  createPoem(poemName: string, poemText: string, generated: boolean):
+      Observable<CreatePoemResponse> {
+    if (!this.user.length) {
+      throw new Error('No user to create poem for');
+    }
+
+    const newPoem: Poem = {
+      'id': Math.max(...this.poem.map(p => p.id)) + 1,
```

Comment:
It probably makes more sense to let the backend/database handle the assignment of IDs if possible.

Reply:
Another note about this - it may make sense for the route for getting/updating to use the ID. If there isn't an ID it would hand the ID-less poem to the backend, and the ID would be assigned and handed back.

Reply:
The database handles ID assignment for new poems/users. I'll include the poem ID in the route for editting.